toplevel/coqtop.ml

```diff
@@ -133,6 +133,20 @@ let print_style_tags opts =
   let () = List.iter iter tags in
   flush_all ()
 
+let ltac_debug_parse () =
+  match DebugHook.Action.parse (read_line ()) with
+  | Ok act -> act
+  | Error error ->
+    CErrors.user_err Pp.(str "ltac_debug_input: " ++ str error)
+
+let ltac_debug_answer = let open DebugHook.Answer in function
+    | Prompt _ -> ()
+    | Goal _ -> ()
+    | Output _ -> ()
+      (* print_notice = Feedback.msg_notice;
+       * print_debug  = Feedback.msg_debug;
+       * print_prompt = Feedback.msg_prompt }); *)
```

Comment:
```suggestion
    | Prompt pp -> Feedback.msg_notice pp
    | Goal pp -> Feedback.msg_notice pp
    | Output pp -> Feedback.msg_notice pp
```

This gets the coqtop debugger to work.

I see the build fails with

```
File "vernac/debugHook.mli", line 26, characters 36-44:
Error: Unbound module Result
```

but oddly enough `make` works fine for me locally.

Reply:
Indeed this wasn't complete, fixed, but used direct terminal access.

The point of this code is actually to access the terminal directly, not use feedback which knows nothing about the output backend, so we can have more control including interactive stuff / readline if desired.

Reply:
Oh, this is due to OCaml 4.05 not having this module, let me fix.